src/core/metallicroughnessmaterial.cpp

```diff
@@ -816,6 +830,22 @@ void MetallicRoughnessMaterial::setToneMappingAlgorithm(MetallicRoughnessEffect:
     m_effect->setToneMappingAlgorithm(algorithm);
 }
 
+void MetallicRoughnessMaterial::onAddedToEntity(Qt3DCore::QEntity *entity)
+{
+    if (m_effect->brdfLUT() != nullptr)
+        return;
+
+    // Looks on the entity tree until we find a SceneEntity
+    // while we find the SceneEntity, get the brdfLut texture and add it to the effect
+    auto *sceneEntity = Kuesa::findParentSceneEntity(entity);
+    if (!sceneEntity) {
+        qCWarning(kuesa) << "A material has been added to a subtree without a SceneEntity";
+        return;
+    }
+
+    m_effect->setBrdfLUT(sceneEntity->texture(QLatin1String("_kuesa_brdfLUT")));
```

Comment:
since we're have access to the SceneEntity here, could we have the texture as one of it's properties and avoid having it in the collection?

Reply:
Modified to use the textuer as a property, instead of adding it to the texture collection